### ISSUE TITLE
Add indicators for sensors (enhancement)

### ISSUE THREAD

**Issue by motobird** [Sublimis/UrbanBiker#20]
Currently, the indicator on the status bar applies generically to all sensors. In other words, when it's red, we only know 1 (or more) sensors is/are not working, but not knowing which one. It's be extremely helpful to have individual indicator for each sensor. For example, when my speed sensor was not communicating, I did not know until I started moving.

**Comment by Sublimis** [comment]
This would probably swarm the status bar too much. General idea was that when a user sees red (or yellow) sensor indicator, they should go to Sensors screen to troubleshoot the problem. Even if the status bar showed which sensor stopped communicating, they would still have to go to Sensors screen to see if this was due to empty battery for example, or the sensor is broken, or it's just reconnecting. Status bar is not suitable for all those information, so we simplified things.

One thing we could do, though, is give an option to show some kind of pop-up dialog and/or produce a sound when a sensor disconnects, which could also simplify opening up the Sensors screen and troubleshooting. It's on our to-do list (ETA undetermined).

**Comment by Sublimis** [comment]
One more thing, most sensors will disconnect after some inactivity period to conserve battery (anywhere from 30 seconds to several minutes, depending on sensor model). Unfortunately, the app cannot differentiate between a broken sensor and a sensor disconnected due to inactivity. Is it because of the inactivity that your speed sensor stopped communicating in the first place?

In any case, as long as tracking is active, the app tries to reconnect all disconnected sensors. Average reconnect interval is 15 seconds. If the sensors icon turns red after you stopped moving, it might be a good idea to first try awakening the sensor by turning the wheel (or pedals if it's combined cadence/speed sensor) and see if the icon turns green in the next 15-20 seconds or so before resuming the ride. Only if it's not turning green for more than 30 seconds, you should troubleshoot.

**Comment by motobird** [comment]
I agree that the status bar should not be too crowded :-) What I meant was to have a way to tell which sensor was not communicating to the app. For ex, I have a 'cadence + speed combo sensor'. In the app sensor page, it showed as green. However, only the cadence was working, but the speed was not, and there was no way to tell until I spun the wheel and saw the speed stayed at 0.  At least, for combo sensor, can the sensor page display the status of cadence and speed status separately?  Urban Biker is an excellent app, if you include turn-by-turn voice guidance to follow a GPX or saved route in the future, you will beat everybody else for sure :-)

**Comment by Sublimis** [comment]
The algorithm for this was improved over the last few months. The status bar "sensors" icon should now stay orange for speed+cadence combo sensor, if the sensor gets connected but no speed or cadence is detected. This will also happen after some timeout period if the sensor drops out during a ride (i.e. no data from the sensor was received for some time, because the magnet fell off or something). Let us know if this is okay now. Thanks for the feedback!
